File: app/src/main/java/com/android/example/instaclone/StartActivity.java
package com.android.example.instaclone;

import android.content.Context;
import android.os.Bundle;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;

import androidx.appcompat.app.AppCompatActivity;

import com.android.example.instaclone.utils.BottomNavigationhelper;
import com.google.android.material.bottomnavigation.BottomNavigationView;

public class StartActivity extends AppCompatActivity {
    private static final String TAG = StartActivity.class.getName();
    private final Context mContext = this;
    private static final int ACTIVITY_NUM = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_start);
        setupBottomNavigationView();
    }
    private void setupBottomNavigationView(){
        Log.d(TAG, "setupBottomNavigationView: setting up BottomNavigationView");
        BottomNavigationView bottomNavigationViewEx = findViewById(R.id.bottom_nav_bar);
        BottomNavigationhelper.setUpBottomNavigation(bottomNavigationViewEx);
        BottomNavigationhelper.enableNavigation(mContext, this,bottomNavigationViewEx);
        Menu menu = bottomNavigationViewEx.getMenu();
        MenuItem menuItem = menu.getItem(ACTIVITY_NUM);
        menuItem.setChecked(true);
    }
}

File: app/src/main/java/com/android/example/instaclone/StartActivity.kt
package com.android.example.instaclone

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.setupWithNavController
import com.google.android.material.bottomnavigation.BottomNavigationView

private lateinit var bottomNav: BottomNavigationView
private lateinit var appBarConfiguration: AppBarConfiguration

//private lateinit var navController: NavController;
class StartActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_start)

        bottomNav = findViewById(R.id.bottom_nav_bar)
        val navHostFragment = supportFragmentManager.findFragmentById(R.id.fragment_container_view_tag) as NavHostFragment
        val navController = navHostFragment.navController
//        navController = findNavController(R.id.fragment_container_view_tag);
        appBarConfiguration = AppBarConfiguration(setOf(R.id.fragment_home, R.id.fragment_search, R.id.postActivity, R.id.fragment_notification, R.id.fragment_profile))
        bottomNav.setupWithNavController(navController)
//        bottomNav.
    }
}